src/object.d

```diff
@@ -2928,7 +2928,15 @@ any GC memory.
 */
 void destroy(T)(T obj) if (is(T == class))
 {
-    rt_finalize(cast(void*)obj);
+    static if(__traits(getLinkage, T) == "C++")
+    {
+        obj.__dtor();
+
+        enum classSize = __traits(classInstanceSize, T);
+        (cast(void*)obj)[0 .. classSize] = typeid(T).initializer[];
```

Comment:
Will this work correctly if the initializer is an array with a null pointer and a non-0 length?

[Function `object.TypeInfo.initializer`](https://dlang.org/library/object/type_info.initializer.html)
>Return default initializer. **If the type should be initialized to all zeros, an array with a null ptr and a length equal to the type size will be returned.** For static arrays, this returns the default initializer for a single element of the array, use tsize to get the correct size.